spec/models/vm_or_template_spec.rb

```diff
@@ -1,6 +1,87 @@
 require "spec_helper"
 
 describe VmOrTemplate do
+  context ".post_refresh_ems" do
+    before do
+      #          fa
+      #       fb
+      #    fc
+      # vc
+      #
+      @before_time = Time.now.utc
+
+      _guid, _server, zone = EvmSpecHelper.local_guid_miq_server_zone
+      @ems      = FactoryGirl.create(:ems_vmware, :zone => zone)
+      @vm       = FactoryGirl.create(:vm_vmware,  :ext_management_system => @ems)
+      @folder_a = FactoryGirl.create(:ems_folder, :ext_management_system => @ems)
+      @folder_b = FactoryGirl.create(:ems_folder, :ext_management_system => @ems)
+      @folder_c = FactoryGirl.create(:ems_folder, :ext_management_system => @ems)
+      @ems.add_child(@folder_a)
+      @folder_a.add_child(@folder_b)
+      @folder_b.add_child(@folder_c)
+      @folder_c.add_child(@vm)
+
+      # post_refresh_ems checks if updated_on >= a provided start time, so jump
+      # forward to avoid subsecond rounding issues.
+      Timecop.travel(1.second)
+      @after_time = Time.now.utc
+    end
+
+    after do
+      Timecop.return
+    end
+
+    def assert_queued_method_for_vm(vm_id, method_name)
+      expected_attributes = {:class_name  => "VmOrTemplate"}
+      if vm_id.kind_of?(Array)
+        expected_attributes[:args] = vm_id
+      else
+        expected_attributes[:instance_id] = vm_id
+      end
```

Comment:
This conditional feels weird from a test perspective...I wonder if it should just be two separate assert_* methods